File: Capítulo_7/ECMAScript 2015 – a classe Set/Operações Set ES2015/Simulando a operação de união/union.js
// Usaremos os dois conjuntos a seguir em nossos exemplos:
const setA = new Set();
setA.add(1);
setA.add(2);
setA.add(3);
const setB = new Set();
setB.add(2);
setB.add(3);
setB.add(4);

const union = (set1, set2) => {
    const unionAb = new Set();
    set1.forEach(value => unionAb.add(value));
    set2.forEach(value => unionAb.add(value));
    return unionAb;
}

console.log(union(setA, setB)); // Set(4) { 1, 2, 3, 4 }

File: Capítulo_7/ECMAScript 2015 – a classe Set/Operações Set ES2015/Simulando a operação de união/Código em Python/union.py
# Usaremos os dois conjuntos a seguir em nossos exemplos:
setA = set()
setA.add(1)
setA.add(2)
setA.add(3)
setB = set()
setB.add(2)
setB.add(3)
setB.add(4)

def union(set1, set2):
    union_ab = set()
    for value in set1:
        union_ab.add(value)
    for value in set2:
        union_ab.add(value)
    return union_ab

union_set = union(setA, setB)

print(union_set)  # {1, 2, 3, 4, 5}